internal/store/maintenance.go

```diff
@@ -83,3 +88,14 @@ func (s *Store) MaintenanceWindow(projectID string) (*atlas.MaintenanceWindow, e
 		return nil, fmt.Errorf("unsupported service: %s", s.service)
 	}
 }
+
+// CreateMaintenanceWindow encapsulates the logic to manage different cloud providers
+func (s *Store) CreateMaintenanceWindow(projectID string, maintenanceWindow *opsmngr.MaintenanceWindow) (*opsmngr.MaintenanceWindow, error) {
```

Comment:
I'm not sure what your plane is when we get to do get/update/rm for ops manager given their signature will probably be different so I'm not sure we should keep them in the same file

Reply:
I was thinking about calling the interface like "OpsManagerMaintenanceWindowsUpdater". Would you rather have one file or two?

Reply:
if you're going to have `OpsManagerMaintenanceWindowUpdater`, we should have `OpsManagerMaintenanceWindowCreator`, I would also make the methods explicit
`CreateOpsManagerMaintenanceWindow` and UpdateOpsManagerMaintenanceWindow` to avoid confusion
separate files is up to you so far I don't have strong opinion on it specially since files within the same package are just symbolic, my only comment is on the name of the interfaces and methods, again to avoid confusion 
Feel free to address any of this in a separate PR